### PR TITLE
Feature/update readme and add java runtime

### PR DESCRIPTION
**Pull request by pmalarme**
This pull request includes significant updates to the `README.md` file, deployment scripts, and infrastructure definitions to enhance the AI Shop project. The changes provide a more comprehensive overview of the project, streamline the deployment process, and improve the structure and readability of the infrastructure code.

### Documentation Updates:
* [`README.md`](diffhunk://#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5L1-R64): Updated project description, added detailed instructions for running locally and deploying to Azure, and included functional and Azure architecture diagrams.
* [`src/ai-image-processing-service/README.md`](diffhunk://#diff-29bfebf051e2581119c5f43637bfa28e2c182678c7e27312569a4ec494c4e762L1-R56): Added instructions for running the service locally and building/pushing Docker images.
* [`src/ai-shop-ui/README.md`](diffhunk://#diff-cfba88d6695ec17d5adccbe1198e0387fa5854c06bca7314d877f59965fa0e90R1-R46): Added instructions for running the UI locally and building/pushing Docker images.
* [`src/api-gateway/README.md`](diffhunk://#diff-041b768310d299c17d199a8f19a1f9ad16b86f32ab00e7536a10fda06864ef97R1-R44): Added instructions for running the API Gateway locally and building/pushing Docker images.

### Deployment Script Enhancements:
* [`deploy.sh`](diffhunk://#diff-30883d1df2fc62f59d66255a70dc05f60d3d4ef4d3957e6e3e38e62f6e471bd0R63-R74): Added commands to retrieve container app names and update their runtime to Java with metrics enabled.

### Infrastructure Code Improvements:
* `infra/deploy.bicep`: 
  * Added detailed parameter descriptions and new parameters for various resources.
  * Reorganized resource definitions for better readability.
  * Added outputs for container app names. [[1]](diffhunk://#diff-5188182d0a83d85f82ddf85d7dcb6c45a51aa98b8b4d3b638e52dd13d2c54f9aR3-R73) [[2]](diffhunk://#diff-5188182d0a83d85f82ddf85d7dcb6c45a51aa98b8b4d3b638e52dd13d2c54f9aR92-R97) [[3]](diffhunk://#diff-5188182d0a83d85f82ddf85d7dcb6c45a51aa98b8b4d3b638e52dd13d2c54f9aR177-R199) [[4]](diffhunk://#diff-5188182d0a83d85f82ddf85d7dcb6c45a51aa98b8b4d3b638e52dd13d2c54f9aL153-R225) [[5]](diffhunk://#diff-5188182d0a83d85f82ddf85d7dcb6c45a51aa98b8b4d3b638e52dd13d2c54f9aR248-R249) [[6]](diffhunk://#diff-5188182d0a83d85f82ddf85d7dcb6c45a51aa98b8b4d3b638e52dd13d2c54f9aR293-R294) [[7]](diffhunk://#diff-5188182d0a83d85f82ddf85d7dcb6c45a51aa98b8b4d3b638e52dd13d2c54f9aL260-R325) [[8]](diffhunk://#diff-5188182d0a83d85f82ddf85d7dcb6c45a51aa98b8b4d3b638e52dd13d2c54f9aR696-R711)

### Minor Fixes:
* [`src/api-gateway/test.http`](diffhunk://#diff-d85fd5948c4687df004be4837777dd78225e967c3dd66293795fcac3fcac9b28L5-R8): Corrected the filename in the multipart form data for testing the API Gateway.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,32 +1,69 @@
-# Used  - An AI powered app, that automtically generates posts on selling platforms. All you need to take is a picture.
+# Java AI - AI Shop
 
-Run the sample locally or within GitHub CodeSpaces.
```
Comment (SandraAhlgrimm) [inline README.md:3]: Please add back the CodeSpaces run, to enable developers to try it on the fly also with flacky internet connection

File: README.md [hunk 2]
```diff
@@ -1,32 +1,69 @@
...
+- `src/api-gateway`: The API Gateway of the application. It is a Spring Boot application that orchestrates the calls to the other services. It uses Spring Cloud Eureka Server for service discovery.
+- `src/eureka-server`: The Eureka Server of the application. It is simple Eureka Server that is only required for local development.
+- `src/ai-image-processing-service`: The AI Image Processing Service of the application. It is a Spring Boot application that uses `Spring AI` to generate the product information based on the image.
+- `src/blob-storage-service`: The Blob Storage Service of the application. It is a Spring Boot application that uses `Spring Cloud Azure` to store the images in Azure Blob Storage. It also provide the generation of the image URL with a SAS token for the `ai-image-processing-service`.
+- `src/item-category-service`: The Item Category Service of the application. It is a Spring Boot application that provides the categories of the items that can be sold in the shop. It uses `Spring Data JPA` to store the categories in a `H2` database. It is infused with AI and uses `LangChain4J` to generate the categories based on the product description.
```
Comment (SandraAhlgrimm) [inline README.md:11]: it's LangChain4j

```suggestion
- `src/item-category-service`: The Item Category Service of the application. It is a Spring Boot application that provides the categories of the items that can be sold in the shop. It uses `Spring Data JPA` to store the categories in a `H2` database. It is infused with AI and uses `LangChain4j` to generate the categories based on the product description.
```

File: media/functional-architecture.png
Comment (SandraAhlgrimm) [inline media/functional-architecture.png:1]: Let's update this and fix the LangChain4J typo. I updated the PowerPoint slide already

File: src/item-category-service/README.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
+# Item Category Service
+
+This service is responsible for providing the categories of the items that can be sold in the shop. It is a RESTful service that returns the categories based on the product description. The service uses the LangChain4J library to generate the categories based on the product information.
```
Comment (SandraAhlgrimm) [inline src/item-category-service/README.md:3]:
```suggestion
This service is responsible for providing the categories of the items that can be sold in the shop. It is a RESTful service that returns the categories based on the product description. The service uses the LangChain4j library to generate the categories based on the product information.
```

File: src/item-category-service/README.md [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+## Run locally
+
+To run the Item Category Service locally, you need first to build the common module and then start the service. Follow the instructions below in the order they are presented:
+
+1. [Build the common module](../java-ai-common/common/README.md)
+2. Ensure that the Eurka Server is running. If not, [start the Eureka Server](../eureka-server/README.md)
+3. Set the following environment variables:
```
Comment (SandraAhlgrimm) [inline src/item-category-service/README.md:11]:
```suggestion
3. Set the following environment variables
```

File: src/item-category-service/README.md [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+    ```
+4. Start the Item Category Service
+
+    ```bash
+    ./mvnw spring-boot:run
+    ```
+
+## Build the image and push it to Azure Container Registry
+
+1. Set the following environment variables:
```
Comment (SandraAhlgrimm) [inline src/item-category-service/README.md:26]:
```suggestion
1. Set the following environment variables
```

File: src/blob-storage-service/README.md [hunk 6]
```diff
@@ -1,3 +1,49 @@
...
+
+3. Start the Blob Storage Service
+
+    ```bash
+    ./mvnw spring-boot:run
+    ```
+
+## Build the image and push it to Azure Container Registry
+
+1. Set the following environment variables:
```
Comment (SandraAhlgrimm) [inline src/blob-storage-service/README.md:26]:
```suggestion
1. Set the following environment variables
```

File: src/java-ai-common/common/README.md [hunk 7]
```diff
@@ -0,0 +1,7 @@
+# AI Shop - Common Module
+
+To build the common module, run the following command:
```
Comment (SandraAhlgrimm) [inline src/java-ai-common/common/README.md:3]:
```suggestion
To build the common module, run the following command
```

File: src/ai-image-processing-service/README.md [hunk 8]
```diff
@@ -1,8 +1,56 @@
...
+
+4. Start the AI Image Processing Service
+
+    ```bash
+    ./mvnw spring-boot:run
+    ```
+
+## Build the image and push it to Azure Container Registry
+
+1. Set the following environment variables:
```
Comment (SandraAhlgrimm) [inline src/ai-image-processing-service/README.md:33]:
```suggestion
1. Set the following environment variables
```

File: src/ai-image-processing-service/README.md [hunk 9]
```diff
@@ -1,8 +1,56 @@
...
+
+The service uses the Azure OpenAI API GPT-4o to generate the information based on the image. The information is then returned to the caller.
+
+## Run locally
+
+To run the AI Image Processing Service locally, you need first to build the common module and then start the service. Follow the instructions below in the order they are presented:
+
+1. [Build the common module](../java-ai-common/common/README.md)
+2. Ensure that the Eurka Server is running. If not, [start the Eureka Server](../eureka-server/README.md)
+3. Set the following environment variables:
```
Comment (SandraAhlgrimm) [inline src/ai-image-processing-service/README.md:19]:
```suggestion
3. Set the following environment variables
```

**Review by SandraAhlgrimm — CHANGES_REQUESTED** [review]
Thanks for all the great work. Especially adding the metrics and the instructions to run.

Just two things: Let's keep the CodeSpaces related instructions. I can add a screenshot how to start if to make it more align with the rest of the instructions.

Let's update LangChain4J to LangChain4j as that's its name. Please be consistent with the usage of new line or : and new line. I added suggestions for those small fixes for your convenience.

**Review by cedricve — APPROVED** [review]
Liuks guut

**Pull request merged**